inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp

```diff
@@ -0,0 +1,538 @@
+    size_t nDims = srcDims.ndims();
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = false;
+    config.inConfs.resize(getParentEdges().size());
+    config.inConfs[DATA_ID].inPlace = -1;
+    config.inConfs[BEGIN_ID].inPlace = -1;
+    config.inConfs[END_ID].inPlace = -1;
+    config.inConfs[DATA_ID].constant = false;
+    config.inConfs[BEGIN_ID].constant = true;
+    config.inConfs[END_ID].constant = true;
+    if (hasStrides) {
+        config.inConfs[STRIDE_ID].inPlace = -1;
+        config.inConfs[STRIDE_ID].constant = true;
+    }
+    config.outConfs.resize(1);
+
+    auto pushSupportedPrimitiveDescriptor = [&](const mkldnn::memory::format_tag inMemoryFormat, const mkldnn::memory::format_tag outMemoryFormat) {
+        config.inConfs[DATA_ID].desc = MKLDNNMemoryDesc(getParentEdgeAt(DATA_ID)->getDims(), dataType, inMemoryFormat);
+        config.inConfs[BEGIN_ID].desc = MKLDNNMemoryDesc(getParentEdgeAt(BEGIN_ID)->getDims(), beginDataType, mkldnn::memory::format_tag::x);
+        config.inConfs[END_ID].desc = MKLDNNMemoryDesc(getParentEdgeAt(END_ID)->getDims(), endDataType, mkldnn::memory::format_tag::x);
+        if (hasStrides)
+            config.inConfs[STRIDE_ID].desc = MKLDNNMemoryDesc(getParentEdgeAt(STRIDE_ID)->getDims(),
+                                                              MKLDNNExtensionUtils::IEPrecisionToDataType(stridePrecision),
+                                                              mkldnn::memory::format_tag::x);
+
+        config.outConfs[0].desc = MKLDNNMemoryDesc(dstDims, dataType, outMemoryFormat);
+        supportedPrimitiveDescriptors.push_back({config, impl_desc_type::unknown, outMemoryFormat});
+    };
+
+    auto canUseBlocked = [=](const size_t blockSize) {
+        return srcDims[1] % blockSize == 0 && abs(stride[1]) == 1 && (begin[1] > srcDims[1] || begin[1] % blockSize == 0);
+    };
+
+    if (nDims == 4 && params.equalDims) {
+        pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nhwc, mkldnn::memory::format_tag::nhwc);
+        if (canUseBlocked(8))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nChw8c, mkldnn::memory::format_tag::nChw8c);
+        if (canUseBlocked(16))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nChw16c, mkldnn::memory::format_tag::nChw16c);
+    } else if (nDims == 5 && params.equalDims) {
+        pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::ndhwc, mkldnn::memory::format_tag::ndhwc);
+        if (canUseBlocked(8))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nCdhw8c, mkldnn::memory::format_tag::nCdhw8c);
+        if (canUseBlocked(16))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nCdhw16c, mkldnn::memory::format_tag::nCdhw16c);
+    }
+    pushSupportedPrimitiveDescriptor(MKLDNNMemory::GetPlainFormat(srcDims), MKLDNNMemory::GetPlainFormat(dstDims));
+}
+
+void MKLDNNStridedSliceNode::createPrimitive() {
```

Comment:
To honest the method looks very complex. It would be great if we can extract its parts into separate functions (DimsNormalization(), DimsGluing(), IndicesCalculation()). Each function should contain description about its purpose (maybe with small example).

Reply:
Added separate functions with small descriptions and examples. Thank you!